kafka-rest/src/main/java/io/confluent/kafkarest/Utils.java

```diff
@@ -43,4 +53,44 @@ public static int errorCodeFromProducerException(Throwable e) {
     }
   }
 
+  public static RestException convertProducerException(KafkaException exception) {
+    if (StringUtil.startsWithIgnoreCase(exception.getMessage(), "Invalid partition")) {
+      throw Errors.partitionNotFoundException();
+    } else if (exception instanceof AuthenticationException) {
+      return Errors.authenticationException(exception.getMessage());
+    } else if (exception instanceof AuthorizationException) {
+      return Errors.authorizationException(exception.getMessage());
+    } else {
+      throw Errors.kafkaErrorException(exception);
+    }
+  }
+
+  public static RestException convertConsumerException(Exception exception) {
+    if (exception != null && !(exception instanceof RestException)) {
```

Comment:
Does this null check make sense? We expect it to be non-null anyway since we have a cast below on the return statement